### PR TITLE
Delete and edit api implementation

### PR DESCRIPTION
**Pull request by Omams21**
I worked on the delete and edit function, where an agent can edit or delete a property posted

### CODE REVIEW

File: src/components/Dashboard/Dashboard.js [hunk 1]
```diff
@@ -53,32 +55,28 @@ export const Dashboard = (props) => {
...
+      <div className="right-container">
+        <p className="header-tag"> Dashboard</p>
+        <div className="welcome-box">
+          <p className="welcome-tag"> Welcome back, Kemi </p>
+          {/* <p> This is your property portfolio report</p> */}
+          {/* <span className="property-report-div">
                   <p className="report-div"> PROPERTY REPORT</p>
                   <p className="report-div"> 63 Vacant</p>
                   <p className="report-div"> 320 Occupied</p>
                 </span> */}
```
Comment (OnafusiMichael) [inline src/components/Dashboard/Dashboard.js]: remove comments

File: src/pages/Signup/Signup.js [hunk 2]
```diff
@@ -1,52 +1,139 @@
...
-                   
-                  
+function Signup() {
+  const navigate = useNavigate();
+  const dispatch = useDispatch();
+  const [firstName, setfirstName] = useState("");
+  const [lastName, setlastName] = useState("");
+  const [password, setpassword] = useState("");
+  const [email, setemail] = useState("");
+  const [phoneNumber, setphoneNumber] = useState("");
```
Comment (OnafusiMichael) [inline src/pages/Signup/Signup.js:16]: I suggest you manage this in one state object

File: src/pages/Uploadproperty/Uploadproperty.js [hunk 3]
```diff
@@ -1,19 +1,72 @@
...
+  const handleChange = (e) => {
+    setInputProperties({ ...inputProperties, [e.target.name]: e.target.value });
+    // console.log(inputProperties);
+  };
+  const handleSubmit = async (e) => {
+    e.preventDefault();
+    // const formData = new FormData();
+
+    // formData.append("file", file);
+    // formData.append("propertyInfo", JSON.stringify(inputProperties));
```
Comment (OnafusiMichael) [inline src/pages/Uploadproperty/Uploadproperty.js]: remove commented lines

**Review by OnafusiMichael — COMMENTED** [review]
This looks good, I suggest you remove all commented lines of code

File: src/App.js [hunk 4]
```diff
@@ -1,56 +1,47 @@
...
-          
-          </Route>
-          <Route path="/sign-in" element={<SignIn/>}>
-          </Route>
+          <Route exact path="/" element={<Homepage />}></Route>
+          <Route path="/sign-up" element={<Signup />}></Route>
+          <Route path="/sign-in" element={<SignIn />}></Route>
 
-    
+          <Route path="/hi" element={<ProtectedRoute />} />
```
Comment (nosisky) [inline src/App.js:27]: What is the use of "hi" route?

File: src/components/Agentprofile/AgentProperties/AgentProperties.jsx [hunk 5]
```diff
@@ -0,0 +1,97 @@
+import React, { useEffect, useState } from 'react'
+import axios from 'axios';
+import { Dashboard } from '../../Dashboard/Dashboard';
+import './style.css'
+import { useParams } from 'react-router-dom';
+import { useNavigate } from 'react-router-dom';
+
+const url = "http://localhost:4000/v1/agent/property";
```
Comment (nosisky) [inline src/components/Agentprofile/AgentProperties/AgentProperties.jsx]: You shouldn't hardcode your API URL. It should be dynamic based on the environment you are on. When you deploy this code online, your API URL will no longer be localhost:3006, how will you solve the issue in that case?

File: src/components/Agentprofile/AgentProperties/AgentProperties.jsx [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+
+    useEffect(()=> {
+        getAgentProperties();
+    }, [])
+
+    const deleteItem = async(id) => {
+      const removeProperty = agentProperties.filter((item) => item.id !== id)
+setAgentProperties(removeProperty)
+      try {
+        const response = await axios.delete(`http://localhost:4000/v1/agent/property/${id}`, config)
```
Comment (nosisky) [inline src/components/Agentprofile/AgentProperties/AgentProperties.jsx]: You shouldn't hardcode your API URL. It should be dynamic based on the environment you are on. When you deploy this code online, your API URL will no longer be localhost:3006, how will you solve the issue in that case?

File: src/components/Dashboard/Dashboard.js [hunk 7]
```diff
@@ -53,32 +55,28 @@ export const Dashboard = (props) => {
...
-
-                </div>
+        </div>
 
+        <div className="New-application-box">
+          <img src="form-icons.png" alt="form" />
+        </div>
       </div>
-
       {props.hello}
```
Comment (nosisky) [inline src/components/Dashboard/Dashboard.js]: What is props.hello?

File: src/components/edit/Edit.js [hunk 8]
```diff
@@ -0,0 +1,255 @@
...
+      // console.log(inputProperties);
+    };
+    const handleSubmit = async (e) => {
+      e.preventDefault();
+      // const formData = new FormData();
+  
+      // formData.append("file", file);
+      // formData.append("propertyInfo", JSON.stringify(inputProperties));
+      try {
+        const response = await axios.put(`http://localhost:4000/v1/agent/property/${id}`, inputProperties, config);
```
Comment (nosisky) [inline src/components/edit/Edit.js]: You shouldn't hardcode your API URL. It should be dynamic based on the environment you are on. When you deploy this code online, your API URL will no longer be localhost:3006, how will you solve the issue in that case?

File: src/pages/Signin/Signin.js [hunk 9]
```diff
@@ -1,44 +1,84 @@
...
+
+const navigate = useNavigate()
+
+  const handleSubmit = async() => {
+   const userInfo = {email, password}
+   const {data} = await axios.post(url,userInfo)
+   const {token} = data
+   localStorage.setItem("token", token)
+  navigate('/dashboard')
+   console.log(token)
```
Comment (nosisky) [inline src/pages/Signin/Signin.js]: Remove this console.log, it's not useful here.

File: src/pages/Signup/Signup.js [hunk 10]
```diff
@@ -1,52 +1,139 @@
-import React from 'react'
-import './signup.css'
+import React, { useState } from "react";
+import "./signup.css";
+import axios from "axios";
+import { useNavigate } from "react-router-dom";
+import { useDispatch } from "react-redux";
+import { setCurrentUser, signupError } from "../../slice/signupslice";
 
+const url = "http://localhost:4000/v1/signup";
```
Comment (nosisky) [inline src/pages/Signup/Signup.js]: You shouldn't hardcode your API URL. It should be dynamic based on the environment you are on. When you deploy this code online, your API URL will no longer be localhost:3006, how will you solve the issue in that case?

File: src/pages/Uploadproperty/Uploadproperty.js [hunk 11]
```diff
@@ -1,19 +1,72 @@
-
 import "./uploadproperty.css";
+import axios from "axios";
+import React, { useState, useEffect } from "react";
+import { Dashboard } from "../../components/Dashboard/Dashboard";
+
+const url = "http://localhost:4000/v1/agent/properties";
```
Comment (nosisky) [inline src/pages/Uploadproperty/Uploadproperty.js]: You shouldn't hardcode your API URL. It should be dynamic based on the environment you are on. When you deploy this code online, your API URL will no longer be localhost:3006, how will you solve the issue in that case?

File: src/components/Agentprofile/AgentProperties/AgentProperties.jsx (hunk 5 above)
Comment (Omams21) [reply src/components/Agentprofile/AgentProperties/AgentProperties.jsx]: I don't understand and how to fix it

**Pull request merged**